src/Property.cpp

```diff
@@ -78,3 +78,21 @@ optional<const Property&> property_map::get(const std::string& name) const {
         return nullopt;
     }
 }
+
+optional<std::unordered_map<std::string, Property>::const_iterator> property_map::cbegin() const {
```

Comment:
I don't think having an optional is really worth it here: it goes against the usual pattern in the STL